### ISSUE TITLE
Error in loading the saved optimizer state

### ISSUE THREAD

**Issue by songofbin** [jiujiezz/deephlapan#9]
Help: please tell me how to solve this problem, thanks!

$ /path/Anaconda2/bin/deephlapan -P LNIMNKLNI -H HLA-A02:01
Using TensorFlow backend.

        =====================================================================
        DeepHLApan is a deep learning approach used for predicting high-confidence 
        neoantigens by considering both the presentation possibilities of 
        mutant peptides and the potential immunogenicity of pMHC.

        Usage:

        Single peptide and HLA:

            deephlapan -P LNIMNKLNI -H HLA-A02:01 

        List of peptides and HLA alleles in a file:

            deephlapan -F [file] -O [output directory]  

            (see 1.csv in demo/ for the detailed format of input file)
        =====================================================================
        
2019-06-06 10:31:48.171028 Prediction starting.....

WARNING:tensorflow:From /path/Anaconda2/lib/python2.7/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
WARNING:tensorflow:From /path/Anaconda2/lib/python2.7/site-packages/Keras-2.2.4-py2.7.egg/keras/backend/tensorflow_backend.py:3445: calling dropout (from tensorflow.python.ops.nn_ops) with keep_prob is deprecated and will be removed in a future version.
Instructions for updating:
Please use `rate` instead of `keep_prob`. Rate should be set to `rate = 1 - keep_prob`.
2019-06-06 10:31:52.526305: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2399935000 Hz
2019-06-06 10:31:52.535517: I tensorflow/compiler/xla/service/service.cc:150] XLA service 0x55c0337b5540 executing computations on platform Host. Devices:
2019-06-06 10:31:52.535933: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (0): <undefined>, <undefined>
WARNING:tensorflow:From /path/Anaconda2/lib/python2.7/site-packages/tensorflow/python/ops/math_ops.py:3066: to_int32 (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
/path/Anaconda2/lib/python2.7/site-packages/Keras-2.2.4-py2.7.egg/keras/engine/saving.py:327: UserWarning: Error in loading the saved optimizer state. As a result, your model is starting with a freshly initialized optimizer.
  warnings.warn('Error in loading the saved optimizer '
2019-06-06 10:36:54.549616 Prediction end

**Comment by songofbin** [comment]
Would you tell me what's the version of Keras you used to train the model?

**Comment by wujingcheng** [comment]
This problem may be caused by the different version of Keras which is 2.0.8 in our training step. You can try to install the same version or use deephlapan by docker.

**Comment by songofbin** [comment]
Thanks!
